PackageIndexer/DotnetPackageIndex.cs

```diff
@@ -119,36 +120,48 @@ private static async Task<IReadOnlyList<PackageIdentity>> GetPackagesFromOtherGa
         return identities.ToArray();
     }
 
-    private static IReadOnlyList<PackageIdentity> GetLatestVersions(IReadOnlyList<PackageIdentity> identities, bool usePreviewVersions)
+    private static IReadOnlyList<PackageIdentity> GetLatestVersions(
+        IReadOnlyList<(PackageIdentity, bool)> identities,
+        bool usePreviewVersions
+        )
     {
         var result = new List<PackageIdentity>();
 
-        IEnumerable<IGrouping<string, PackageIdentity>> groups = identities.GroupBy(i => i.Id);
+        IEnumerable<IGrouping<string, (PackageIdentity, bool)>> groups = identities.GroupBy(i => i.Item1.Id).OrderBy(g => g.Key);
 
-        foreach (IGrouping<string, PackageIdentity> group in groups.OrderBy(g => g.Key))
+        foreach (IGrouping<string, (PackageIdentity, bool)> group in groups)
         {
             string packageId = group.Key;
-            IOrderedEnumerable<PackageIdentity> versions = group.OrderByDescending(p => p.Version, VersionComparer.VersionReleaseMetadata);
+            IOrderedEnumerable<(PackageIdentity, bool)> versions =
+                group.OrderByDescending(p => p.Item1.Version, VersionComparer.VersionReleaseMetadata);
+
+            (PackageIdentity, bool) latestStable = versions.FirstOrDefault(i => !i.Item1.Version.IsPrerelease);
+            (PackageIdentity, bool) latestPrerelease = versions.FirstOrDefault(i => i.Item1.Version.IsPrerelease);
 
-            PackageIdentity latestStable = versions.FirstOrDefault(i => !i.Version.IsPrerelease);
-            PackageIdentity latestPrerelease = versions.FirstOrDefault(i => i.Version.IsPrerelease);
+            // If the latest version is deprecated, don't include any version.
+            if (latestStable.Item2)
+                latestStable = default;
+            if (latestPrerelease.Item2)
+                latestPrerelease = default;
```

Comment:
Here, this might simpler with named tuples:

```suggestion
            if (latestStable.isDeprecated)
                latestStable = default;
            if (latestPrerelease.isDeprecated)
                latestPrerelease = default;
```